rules/manager.go

```diff
@@ -188,13 +214,15 @@ func (m *Manager) Stop() {
 	level.Info(m.logger).Log("msg", "Rule manager stopped")
 }
 
+type AlertStateStoreFunc func(g *Group)
+
 // Update the rule manager's state as the config requires. If
 // loading the new rules failed the old rule set is restored.
-func (m *Manager) Update(interval time.Duration, files []string, externalLabels labels.Labels, externalURL string, groupEvalIterationFunc GroupEvalIterationFunc) error {
+func (m *Manager) Update(interval time.Duration, files []string, externalLabels labels.Labels, externalURL string, groupEvalIterationFunc GroupEvalIterationFunc, alertStateStoreFunc AlertStateStoreFunc) error {
```

Comment:
Can't the `AlertStateStoreFunc` func be part of the `ManagerOptions`?

Or let me ask the question in a different way: any reason why we always need to pass alertStateStoreFunc to `Update` and `LoadGroups`. Will this function vary across invocations?  AlertStateStoreFunc could be part of the manager and you only need to set that during initialization. This way you don't break any existing APIs.

Reply:
It can be part of `ManagerOptions`. 

Agree with the suggestion, will remove it from `Update` and `LoadGroups`.